src/test/kotlin/org/rust/ide/annotator/RsAnnotationTestBase.kt

```diff
@@ -0,0 +1,186 @@
+import org.rust.openapiext.Testmark
+
+abstract class RsAnnotationTestBase : RsTestBase() {
+    protected fun doTest(vararg additionalFilenames: String) {
+        myFixture.testHighlighting(fileName, *additionalFilenames)
+    }
+
+    protected fun checkInfo(@Language("Rust") text: String)  =
+        checkByText(text, checkWarn = false, checkWeakWarn = false, checkInfo = true)
+
+    protected fun checkWarnings(@Language("Rust") text: String) =
+        checkByText(text, checkWarn = true, checkWeakWarn = true, checkInfo = false)
+
+    protected fun checkErrors(@Language("Rust") text: String) =
+        checkByText(text, checkWarn = false, checkWeakWarn = false, checkInfo = false)
+
+    protected fun checkByText(
+        @Language("Rust") text: String,
+        checkWarn: Boolean = true,
+        checkInfo: Boolean = false,
+        checkWeakWarn: Boolean = false,
+        testmark: Testmark? = null
+    ) = check(text,
+        checkWarn = checkWarn,
+        checkInfo = checkInfo,
+        checkWeakWarn = checkWeakWarn,
+        configure = this::configureByText,
+        testmark = testmark)
+
+    protected fun checkFixByText(
+        fixName: String,
+        @Language("Rust") before: String,
+        @Language("Rust") after: String,
+        checkWarn: Boolean = true,
+        checkInfo: Boolean = false,
+        checkWeakWarn: Boolean = false,
+        testmark: Testmark? = null
+    ) = checkFix(fixName, before, after,
+        configure = this::configureByText,
+        checkBefore = { myFixture.checkHighlighting(checkWarn, checkInfo, checkWeakWarn) },
+        checkAfter = this::checkByText,
+        testmark = testmark)
+
+    protected fun checkFixByTextWithoutHighlighting(
+        fixName: String,
+        @Language("Rust") before: String,
+        @Language("Rust") after: String,
+        testmark: Testmark? = null
+    ) = checkFix(fixName, before, after,
+        configure = this::configureByText,
+        checkBefore = {},
+        checkAfter = this::checkByText,
+        testmark = testmark)
+
+    protected fun checkByFileTree(
+        @Language("Rust") text: String,
+        checkWarn: Boolean = true,
+        checkInfo: Boolean = false,
+        checkWeakWarn: Boolean = false,
+        testmark: Testmark? = null
+    ) = check(text,
+        checkWarn = checkWarn,
+        checkInfo = checkInfo,
+        checkWeakWarn = checkWeakWarn,
+        configure = this::configureByFileTree,
+        testmark = testmark)
+
+    protected fun checkFixByFileTree(
+        fixName: String,
+        @Language("Rust") before: String,
+        @Language("Rust") after: String,
+        checkWarn: Boolean = true,
+        checkInfo: Boolean = false,
+        checkWeakWarn: Boolean = false,
+        testmark: Testmark? = null
+    ) = checkFix(fixName, before, after,
+        configure = this::configureByFileTree,
+        checkBefore = { myFixture.checkHighlighting(checkWarn, checkInfo, checkWeakWarn) },
```

Comment:
Why are we running `checkHighlighting` for quickFix tests? Is it true that here we running all annotators here?

Reply:
> Why are we running checkHighlighting for quickFix tests?

It allows us to check inspection message and result of quick fix in one test case. For example, we use it in auto import tests.

> Is it true that here we running all annotators here?

Yeah. But if you don't need highlighting, we have `checkFixByTextWithoutHighlighting` method

Reply:
Okaaaay